### DISCUSSION TITLE
Running the Wio Terminal Nightlight on power up

### DISCUSSION THREAD

**Discussion by marty-optum** [microsoft/IoT-For-Beginners#298] [Q&A]
I was able to get the [Wio Terminal nightlight application](https://github.com/microsoft/IoT-For-Beginners/blob/main/1-getting-started/lessons/3-sensors-and-actuators/wio-terminal-actuator.md) to work with both the sensor and the actuator when it is loaded and invoked through VS Code and the **PlatformIO: Upload** and **PlatformIO: Serial Monitor** commands.

My understand is that once this code is uploaded to the Wio terminal, it stays there in non-volatile memory until it is erased or something else is uploaded in its place.  Is this true?  Or is the nightlight program cleared from memory when the Wio terminal is restarted?

If it stays in non-volatile memory, I was assuming that the nightlight program would start automatically when the Wio Terminal powered up, but this is not happening.  Should it be?

Thanks in advance for your enlightenment :-)

**Comment by jimbobbennett** [comment]
It stays on the flash memory when the device is powered down. I've noticed that connecting the Wio Terminal to a computer won't automatically start the code running until you connect using a serial monitor, but connecting direct to power does start the code running. I still need to dig into the exact scenarios but I think it's trying to be smart and not start your code till you are monitoring it when connected to a computer.
